cohere/responses/generation.py

```diff
@@ -114,3 +115,22 @@ def prompt(self) -> str:
         """Returns the prompt used as input"""
         return self[0].prompt  # should all be the same
 
+
+StreamingTokenLikelihood = NamedTuple("TokenLikelihood", [("index", int), ("token", str), ("likelihood", float)])
```

Comment:
not sure about my python - are these strongly typed fields or are they accessed by name like a dict?